src/pip/_internal/configuration.py

```diff
@@ -168,7 +175,11 @@ def set_value(self, key: str, value: Any) -> None:
                 parser.add_section(section)
             parser.set(section, name, value)
 
-        self._config[self.load_only][key] = value
+        exists = self._config[self.load_only].get(fname)
+        if not exists:
+            self._config[self.load_only][fname] = {}
+
+        self._config[self.load_only][fname][key] = value
```

Comment:
Can this become

```python
config = self._config[self.load_only]
try:
    config[fname][key] = value
except KeyError:
    config[fname] = {key: value}
```

or with `self._config[self.load_only].setdefault(fname, {})`?

Reply:
There’s another `if not exists` case below that may be similarly modified.